File: 2_Add_two_numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* result = new ListNode();
        auto listIterator = result;
        int carry = 0;
        while(l1 != nullptr || l2 != nullptr){
            int x = (l1 != nullptr)?l1->val:0;
            int y = (l2 != nullptr)?l2->val:0;
            
            int sum = x + y + carry;
            int remainder = sum % 10;
            carry = sum / 10;
            
            listIterator->next = new ListNode(remainder);
            listIterator = listIterator->next;
            
            if(l1 != nullptr){
                l1 = l1->next;
            }
            if(l2 != nullptr){
                l2 = l2->next;
            }
        }
        if(carry){
            listIterator->next = new ListNode(carry);
        }
        return result->next;
    }

};

File: 2_Add_two_numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        head = temp = ListNode(0)
        remainder = 0
        while l1 or l2:
            totalVal = 0
            if l1:
                totalVal += l1.val
                l1 = l1.next  
            if l2:
                totalVal += l2.val
                l2 = l2.next
           
            (remainder,val) = divmod(totalVal+remainder, 10)
            temp.next = ListNode(val)
            temp = temp.next
        
        if remainder > 0:
            temp.next = ListNode(remainder)
        
        return head.next